src/main/java/com/dataloom/data/requests/GetEntitySetRequest.java

```diff
@@ -0,0 +1,62 @@
+package com.dataloom.data.requests;
+
+import java.util.Set;
+import java.util.UUID;
+
+import com.dataloom.data.SerializationConstants;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class GetEntitySetRequest {
+    @JsonProperty( SerializationConstants.SYNC_IDS )
+    private Optional<Set<UUID>> syncIds;
+    @JsonProperty( SerializationConstants.PROPERTIES_FIELD )
+    private Optional<Set<UUID>> selectedProperties;
+
+    @JsonCreator
+    public GetEntitySetRequest(
+            @JsonProperty( SerializationConstants.SYNC_IDS ) Optional<Set<UUID>> syncIds,
```

Comment:
Making collection arguments Optional is fine, but internal fields should be an empty Immutable collection if argument is not provided. That is don't create a distinction between empty collection and no data if you don't have to.